cmd/api/logging/logging.go

```diff
@@ -0,0 +1,162 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package logging
+
+import (
+	"bufio"
+	"crypto/tls"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"os"
+
+	"github.com/gin-gonic/gin"
+	"github.com/kubearchive/kubearchive/cmd/api/abort"
+	"github.com/kubearchive/kubearchive/pkg/files"
+	"github.com/ohler55/ojg/jp"
+	"github.com/ohler55/ojg/oj"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+const (
+	userKey     string = "USER"
+	passwordKey string = "PASSWORD"
+)
+
+// GetKubeArchiveLoggingCredentials retrieves the kubearchive-logging secret data and
+// an error if there isn't available
+func GetKubeArchiveLoggingCredentials() (map[string]string, error) {
+	loggingDir, exists := os.LookupEnv(files.LoggingDirEnvVar)
+	if !exists {
+		return nil, fmt.Errorf("environment variable not set: %s", files.LoggingDirEnvVar)
+	}
+	configFiles, err := files.FilesInDir(loggingDir)
+	if err != nil {
+		return nil, fmt.Errorf("could not read logging credentials: %w", err)
+	}
+	if len(configFiles) == 0 {
+		return nil, fmt.Errorf("logging secret is empty. To configure logging update the kubearchive-logging Secret")
+	}
+
+	loggingConf, err := files.LoggingConfigFromFiles(configFiles)
+	if err != nil {
+		return nil, fmt.Errorf("could not get value for logging credentials: %w", err)
+	}
+	return loggingConf, nil
+}
+
+// LogRetrieval retrieves a middleware function that checks the logging config and when set
+// expects to find a log url in the context from where retrieve and parse logs
+func LogRetrieval(loggingCreds map[string]string, loggingCredsErr error) gin.HandlerFunc {
+	user := loggingCreds[userKey]
+	password := loggingCreds[passwordKey]
+	if user == "" || password == "" || loggingCredsErr != nil {
+		loggingCredsErr = fmt.Errorf(
+			"logging secret values unset. To configure logging update the kubearchive-logging Secret")
+		return func(c *gin.Context) {
+			abort.Abort(c, loggingCredsErr, http.StatusBadRequest)
+		}
+	}
+	// FIXME For now the queries to the logging backend server are done insecurely. Needed for the current test env.
+	// #nosec G402
+	client := http.Client{
+		Transport: otelhttp.NewTransport(&http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true}})}
+
+	return func(c *gin.Context) {
+		logUrl := c.GetString("logURL")
+		jsonPath := c.GetString("jsonPath")
+
+		if logUrl == "" {
+			abort.Abort(c, fmt.Errorf("no log URL found"), http.StatusNotFound)
+			return
+		}
+		slog.Info("Retrieving logs", "logURL", logUrl)
+
+		var jsonPathParser jp.Expr
```

Comment:
What happens if this never gets populated? What does it match by default?

Reply:
if this value never gets populated `jsonPathParser` is nil so when the line is parsed in the `parseLine` function (line 128), when `jsonPathParser` is checked, if its value is nil, the whole line is returned without applying any filters.